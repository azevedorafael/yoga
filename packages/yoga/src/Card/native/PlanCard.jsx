import React from 'react';
import styled from 'styled-components';
import Card from './Card';

const Plan = styled(Card)``;

const PlanCard = ({ ...rest }) => <Plan {...rest} />;

PlanCard.displayName = 'PlanCard';

export default PlanCard;
