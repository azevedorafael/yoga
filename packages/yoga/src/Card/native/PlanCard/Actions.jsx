import React from 'react';
import styled from 'styled-components';
import { arrayOf, oneOfType } from 'prop-types';

import Button from '../../../Button';
import { typeOf } from '../../../shared/propTypes';
import Actions from '../Card/Actions';

const Action = styled(Actions)`
  display: flex;
  ${({
    lastChild,
    theme: {
      yoga: { spacing },
    },
  }) => `
    ${!lastChild ? `margin-bottom: ${spacing.xsmall}px;` : ''}
  `}
`;

const ActionWrapper = styled.View`
  ${({
    theme: {
      yoga: { spacing },
    },
  }) => `
    padding-top: ${spacing.xsmall}px;
  `}
`;

const PlanActions = ({ children }) => {
  const actionsLength = React.Children.count(children);
  return (
    <ActionWrapper>
      {React.Children.map(children, (child, index) => (
        <Action lastChild={actionsLength === index + 1}>{child}</Action>
      ))}
    </ActionWrapper>
  );
};

PlanActions.propTypes = {
  children: oneOfType([
    oneOfType([
      typeOf(Button),
      typeOf(Button.Text),
      typeOf(Button.Outline),
      typeOf(Button.Link),
    ]),
    arrayOf(
      oneOfType([
        typeOf(Button),
        typeOf(Button.Text),
        typeOf(Button.Outline),
        typeOf(Button.Link),
      ]),
    ),
  ]),
};

PlanActions.defaultProps = {
  children: null,
};

PlanActions.displayName = 'PlanCard.Actions';

export default PlanActions;
