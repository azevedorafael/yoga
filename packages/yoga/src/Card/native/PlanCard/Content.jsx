import React, { useContext } from 'react';
import styled from 'styled-components';
import { string, node } from 'prop-types';

import Content from '../Card/Content';
import PlanCardContext from './PlanCardContext';

const Title = styled.Text.attrs({
  numberOfLines: 2,
})`
  ${({
    variant,
    theme: {
      yoga: {
        components: {
          card: { plan },
        },
        colors: { white, dark },
      },
    },
  }) => `
  height: 40px;
  margin-top: ${plan.title.margin.top}px;
  margin-bottom: ${plan.title.margin.bottom}px;

  font-size: ${plan.title.font.size}px;
  font-weight: ${plan.title.font.weight};

  color: ${variant ? white : dark};
`}
`;

const Price = styled.Text`
  ${({
    variant,
    theme: {
      yoga: {
        components: {
          card: { plan },
        },
        colors: { white, dark },
      },
    },
  }) => `
  margin-top: ${plan.price.margin.top}px;

  font-size: ${plan.price.font.size}px;
  font-weight: ${plan.price.font.weight};
  color: ${variant ? white : dark};
  `}
`;
const Period = styled.Text`
  ${({
    variant,
    theme: {
      yoga: {
        components: {
          card: { plan },
        },
        colors: { white, dark },
      },
    },
  }) => `
  margin-bottom: ${plan.period.margin.bottom}px;

  font-size: ${plan.period.font.size}px;
  font-weight: ${plan.period.font.weight};
  color: ${variant ? white : dark};
  `}
`;

const PlanCardContent = ({ title, price, period, children }) => {
  const { variant } = useContext(PlanCardContext);

  return (
    <Content>
      {title && <Title variant={variant}>{title}</Title>}
      {price && <Price variant={variant}>{price}</Price>}
      {period && <Period variant={variant}>{period}</Period>}

      {children}
    </Content>
  );
};

PlanCardContent.propTypes = {
  title: string.isRequired,
  price: string.isRequired,
  period: string.isRequired,
  children: node,
};

PlanCardContent.defaultProps = {
  children: null,
};

PlanCardContent.displayName = 'PlanCard.Content';

export default PlanCardContent;
