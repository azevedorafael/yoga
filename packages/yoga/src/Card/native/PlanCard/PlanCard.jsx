import React from 'react';
import styled from 'styled-components';
import { oneOf, shape, string } from 'prop-types';

import Card from '../Card';
import PlanCardContext from './PlanCardContext';

const Plan = styled(Card)`
  max-width: 280px;
  width: 100%;
`;

const PlanCard = ({ variant, variantIntensity, ...rest }) => (
  <PlanCardContext.Provider value={{ variant, variantIntensity }}>
    <Plan {...rest} variant={variant} variantIntensity={variantIntensity} />
  </PlanCardContext.Provider>
);

PlanCard.propTypes = {
  /** text: the content inside the Card Ribbon
   * variant: style the ribbon following the theme (primary, secondary, tertiary) */
  /** variantIntensity: ribbon variant color intensity (0, 1, 2, 3) */
  ribbon: shape({
    text: string,
    variant: oneOf(['', 'primary', 'secondary']),
    variantIntensity: oneOf([0, 1, 2, 3]),
  }),
  /** style the card following the theme (primary, secondary, tertiary) */
  variant: oneOf(['', 'primary', 'secondary']),
  /** intensity of variant color (0, 1, 2, 3) */
  variantIntensity: oneOf([0, 1, 2, 3]),
};

PlanCard.defaultProps = {
  ribbon: {},
  variant: '',
  variantIntensity: 2,
};

PlanCard.displayName = 'PlanCard';

export default PlanCard;
