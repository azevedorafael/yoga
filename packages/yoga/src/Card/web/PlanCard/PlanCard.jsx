import React from 'react';
import styled from 'styled-components';
import { oneOf, shape, string } from 'prop-types';

import Card from '../Card';

const Plan = styled(Card)`
  max-width: 280px;
  ${({
    variant,
    theme: {
      yoga: {
        colors: { white },
      },
    },
  }) => `
  ${variant ? `color: ${white};` : ''}
`}
`;

const PlanCard = ({ ...rest }) => <Plan {...rest} />;

PlanCard.propTypes = {
  /** text: the content inside the Card Ribbon
   * variant: style the ribbon following the theme (primary, secondary, tertiary)
   * variantIntensity: ribbon variant color intensity (0, 1, 2, 3) */
  ribbon: shape({
    text: string,
    variant: oneOf(['', 'primary', 'secondary']),
    variantIntensity: oneOf([0, 1, 2, 3]),
  }),
  /** style the card following the theme (primary, secondary, tertiary) */
  variant: oneOf(['', 'primary', 'secondary']),
  /** intensity of variant color (0, 1, 2, 3) */
  variantIntensity: oneOf([0, 1, 2, 3]),
};

PlanCard.defaultProps = {
  ribbon: {},
  variant: '',
  variantIntensity: 2,
};
PlanCard.displayName = 'PlanCard';

export default PlanCard;
