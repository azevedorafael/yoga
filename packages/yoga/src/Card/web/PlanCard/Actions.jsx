import styled from 'styled-components';
import Actions from '../Card/Actions';

const PlanActions = styled(Actions)`
  display: grid;
  ${({
    theme: {
      yoga: { spacing },
    },
  }) => `
    grid-gap: ${spacing.xsmall}px;
    padding-top: ${spacing.xsmall}px;
  `}
`;

PlanActions.displayName = 'PlanCard.Actions';

export default PlanActions;
