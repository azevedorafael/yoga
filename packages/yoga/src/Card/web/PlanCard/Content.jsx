import React from 'react';
import styled from 'styled-components';
import Content from '../Card/Content';

const Title = styled.h3`
  ${({
    theme: {
      yoga: {
        components: {
          card: { plan },
        },
      },
    },
  }) => `
  display: -webkit-box;
  height: 40px;
  margin: ${plan.title.margin.top}px 0 ${plan.title.margin.bottom}px;

  font-size: ${plan.title.font.size}px;
  font-weight: ${plan.title.font.weight};

  color: inherit;
  -webkit-line-clamp: 2;
  overflow: hidden;
  text-overflow: ellipsis;
  text-overflow: -o-ellipsis-lastline;
`}
`;

const Price = styled.strong`
  display: block;

  ${({
    theme: {
      yoga: {
        components: {
          card: { plan },
        },
      },
    },
  }) => `
  margin-top: ${plan.price.margin.top}px;

  font-size: ${plan.price.font.size}px;
  font-weight: ${plan.price.font.weight};
  `}
`;
const Period = styled.span`
  display: block;

  ${({
    theme: {
      yoga: {
        components: {
          card: { plan },
        },
      },
    },
  }) => `
  display: block;
  margin-bottom: ${plan.period.margin.bottom}px;

  font-size: ${plan.period.font.size}px;
  font-weight: ${plan.period.font.weight};
  `}
`;

const PlanCardContent = ({ title, price, period }) => {
  return (
    <Content>
      {title && <Title>{title}</Title>}
      {price && <Price>{price}</Price>}
      {period && <Period>{period}</Period>}
    </Content>
  );
};

export default PlanCardContent;
