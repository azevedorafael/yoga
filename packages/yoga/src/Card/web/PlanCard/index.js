import Card from '../Card';
import PlanCard from './PlanCard';
import Content from './Content';

PlanCard.Header = Card.Header;
PlanCard.Content = Content;
PlanCard.Actions = Card.Actions;

export default PlanCard;
