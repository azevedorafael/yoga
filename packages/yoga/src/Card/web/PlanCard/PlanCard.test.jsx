import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider, Button } from '../../..';
import PlanCard from '.';

describe('<PlanCard />', () => {
  describe('Snapshots', () => {
    it('should match snapshot with default PlanCard', () => {
      const { container } = render(
        <ThemeProvider>
          <PlanCard>
            <PlanCard.Header>Gympass PlanCard</PlanCard.Header>
            <PlanCard.Content
              title="Gympass Black for employees with Buddha Spa + Reebook included"
              price="U$ 99.90"
              period="/month"
            >
              content&apos;s children is rendered
            </PlanCard.Content>
            <PlanCard.Actions>
              <Button>should have buttons</Button>
            </PlanCard.Actions>
          </PlanCard>
        </ThemeProvider>,
      );
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with secondary PlanCard', () => {
      const { container } = render(
        <ThemeProvider>
          <PlanCard variant="secondary">
            <PlanCard.Header>Gympass PlanCard</PlanCard.Header>
            <PlanCard.Content
              title="Gympass Black for employees with Buddha Spa + Reebook included"
              price="U$ 99.90"
              period="/month"
            >
              content&apos;s children is rendered
            </PlanCard.Content>
            <PlanCard.Actions>
              <Button>should have buttons</Button>
            </PlanCard.Actions>
          </PlanCard>
        </ThemeProvider>,
      );
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with tertiary PlanCard', () => {
      const { container } = render(
        <ThemeProvider>
          <PlanCard variant="tertiary">
            <PlanCard.Header>Gympass PlanCard</PlanCard.Header>
            <PlanCard.Content
              title="Gympass Black for employees with Buddha Spa + Reebook included"
              price="U$ 99.90"
              period="/month"
            >
              content&apos;s children is rendered
            </PlanCard.Content>
            <PlanCard.Actions>
              <Button>should have buttons</Button>
            </PlanCard.Actions>
          </PlanCard>
        </ThemeProvider>,
      );
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot PlanCard with ribbon', () => {
      const { container } = render(
        <ThemeProvider>
          <PlanCard
            ribbon={{
              text: 'Label',
            }}
          />
        </ThemeProvider>,
      );
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot PlanCard with secondary ribbon', () => {
      const { container } = render(
        <ThemeProvider>
          <PlanCard
            ribbon={{
              text: 'Label',
              variant: 'secondary',
            }}
          />
        </ThemeProvider>,
      );
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot PlanCard with tertiary ribbon', () => {
      const { container } = render(
        <ThemeProvider>
          <PlanCard
            ribbon={{
              text: 'Label',
              variant: 'tertiary',
            }}
          />
        </ThemeProvider>,
      );
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot default PlanCard with Content', () => {
      const { container } = render(
        <ThemeProvider>
          <PlanCard>
            <PlanCard.Content
              title="Gympass Black for employees with Buddha Spa + Reebook included"
              price="U$ 99.90"
              period="/month"
            />
          </PlanCard>
        </ThemeProvider>,
      );
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot variant PlanCard with Content', () => {
      const { container } = render(
        <ThemeProvider>
          <PlanCard variant="secondary">
            <PlanCard.Content
              title="Gympass Black for employees with Buddha Spa + Reebook included"
              price="U$ 99.90"
              period="/month"
            />
          </PlanCard>
        </ThemeProvider>,
      );
      expect(container).toMatchSnapshot();
    });
  });
});
