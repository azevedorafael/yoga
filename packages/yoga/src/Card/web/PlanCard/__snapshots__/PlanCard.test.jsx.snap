// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlanCard /> Snapshots should match snapshot PlanCard with ribbon 1`] = `
.c0 {
  padding: 16px  16px  16px  16px;
  border-radius: 8px;
  background-color: #FFFFFF;
  box-shadow: 0 2px 6px rgba(152,152,166,0.25);
}

.c2 {
  display: inline-block;
  top: 16px;
  left: 0;
  margin-left: -16px;
  margin-bottom: 16px;
  padding: 4px 12px 4px 16px;
  border-top-right-radius: 9999px;
  border-bottom-right-radius: 9999px;
  background-color: #F5F5FA;
  color: #41414A;
}

.c1 {
  max-width: 280px;
  width: 100%;
}

<div>
  <section
    class="c0 c1"
  >
    <span
      class="c2"
    >
      Label
    </span>
  </section>
</div>
`;

exports[`<PlanCard /> Snapshots should match snapshot PlanCard with secondary ribbon 1`] = `
.c0 {
  padding: 16px  16px  16px  16px;
  border-radius: 8px;
  background-color: #FFFFFF;
  box-shadow: 0 2px 6px rgba(152,152,166,0.25);
}

.c2 {
  display: inline-block;
  top: 16px;
  left: 0;
  margin-left: -16px;
  margin-bottom: 16px;
  padding: 4px 12px 4px 16px;
  border-top-right-radius: 9999px;
  border-bottom-right-radius: 9999px;
  background-color: #652CDB;
  color: #FFFFFF;
}

.c1 {
  max-width: 280px;
  width: 100%;
}

<div>
  <section
    class="c0 c1"
  >
    <span
      class="c2"
    >
      Label
    </span>
  </section>
</div>
`;

exports[`<PlanCard /> Snapshots should match snapshot PlanCard with tertiary ribbon 1`] = `
.c0 {
  padding: 16px  16px  16px  16px;
  border-radius: 8px;
  background-color: #FFFFFF;
  box-shadow: 0 2px 6px rgba(152,152,166,0.25);
}

.c2 {
  display: inline-block;
  top: 16px;
  left: 0;
  margin-left: -16px;
  margin-bottom: 16px;
  padding: 4px 12px 4px 16px;
  border-top-right-radius: 9999px;
  border-bottom-right-radius: 9999px;
  background-color: #2B9486;
  color: #FFFFFF;
}

.c1 {
  max-width: 280px;
  width: 100%;
}

<div>
  <section
    class="c0 c1"
  >
    <span
      class="c2"
    >
      Label
    </span>
  </section>
</div>
`;

exports[`<PlanCard /> Snapshots should match snapshot default PlanCard with Content 1`] = `
.c0 {
  padding: 16px  16px  16px  16px;
  border-radius: 8px;
  background-color: #FFFFFF;
  box-shadow: 0 2px 6px rgba(152,152,166,0.25);
}

.c1 {
  max-width: 280px;
  width: 100%;
}

.c2 {
  display: -webkit-box;
  height: 40px;
  margin: 8px 0 12px;
  font-size: 14px;
  font-weight: 400;
  color: inherit;
  -webkit-line-clamp: 2;
  overflow: hidden;
  text-overflow: ellipsis;
  text-overflow: -o-ellipsis-lastline;
}

.c3 {
  display: block;
  margin-top: 12px;
  font-size: 20px;
  font-weight: 700;
}

.c4 {
  display: block;
  margin-bottom: 8px;
  font-size: 14px;
  font-weight: 400;
}

<div>
  <section
    class="c0 c1"
  >
    <div
      class=""
    >
      <h3
        class="c2"
      >
        Gympass Black for employees with Buddha Spa + Reebook included
      </h3>
      <strong
        class="c3"
      >
        U$ 99.90
      </strong>
      <span
        class="c4"
      >
        /month
      </span>
    </div>
  </section>
</div>
`;

exports[`<PlanCard /> Snapshots should match snapshot variant PlanCard with Content 1`] = `
.c0 {
  padding: 16px  16px  16px  16px;
  border-radius: 8px;
  background-color: #FFFFFF;
  box-shadow: 0 2px 6px rgba(152,152,166,0.25);
}

.c1 {
  max-width: 280px;
  width: 100%;
  color: #FFFFFF;
  background-color: #652CDB;
}

.c2 {
  display: -webkit-box;
  height: 40px;
  margin: 8px 0 12px;
  font-size: 14px;
  font-weight: 400;
  color: inherit;
  -webkit-line-clamp: 2;
  overflow: hidden;
  text-overflow: ellipsis;
  text-overflow: -o-ellipsis-lastline;
}

.c3 {
  display: block;
  margin-top: 12px;
  font-size: 20px;
  font-weight: 700;
}

.c4 {
  display: block;
  margin-bottom: 8px;
  font-size: 14px;
  font-weight: 400;
}

<div>
  <section
    class="c0 c1"
  >
    <div
      class=""
    >
      <h3
        class="c2"
      >
        Gympass Black for employees with Buddha Spa + Reebook included
      </h3>
      <strong
        class="c3"
      >
        U$ 99.90
      </strong>
      <span
        class="c4"
      >
        /month
      </span>
    </div>
  </section>
</div>
`;

exports[`<PlanCard /> Snapshots should match snapshot with default PlanCard 1`] = `
.c6 {
  box-sizing: border-box;
  outline: none;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  cursor: pointer;
  width: auto;
  height: 48px;
  padding-left: 24px;
  padding-right: 24px;
  background-color: #F46152;
  border: 2px solid #F46152;
  border-radius: 9999px;
  color: #FFFFFF;
  font-size: 14px;
  font-weight: 700;
}

.c6:not([disabled]):hover,
.c6:not([disabled]):focus {
  box-shadow: 0 4px 8px rgba(244,97,82,0.45);
}

.c6:active {
  background-color: #F78D82;
  border-color: #F78D82;
  color: #FFFFFF;
}

.c6:disabled {
  background-color: #DBDBE8;
  border-color: #DBDBE8;
  color: #9898A6;
  cursor: not-allowed;
}

.c0 {
  padding: 16px  16px  16px  16px;
  border-radius: 8px;
  background-color: #FFFFFF;
  box-shadow: 0 2px 6px rgba(152,152,166,0.25);
}

.c1 {
  max-width: 280px;
  width: 100%;
}

.c2 {
  display: -webkit-box;
  height: 40px;
  margin: 8px 0 12px;
  font-size: 14px;
  font-weight: 400;
  color: inherit;
  -webkit-line-clamp: 2;
  overflow: hidden;
  text-overflow: ellipsis;
  text-overflow: -o-ellipsis-lastline;
}

.c3 {
  display: block;
  margin-top: 12px;
  font-size: 20px;
  font-weight: 700;
}

.c4 {
  display: block;
  margin-bottom: 8px;
  font-size: 14px;
  font-weight: 400;
}

.c5 {
  display: grid;
  grid-gap: 8px;
  padding-top: 8px;
}

<div>
  <section
    class="c0 c1"
  >
    <header
      class=""
    >
      Gympass PlanCard
    </header>
    <div
      class=""
    >
      <h3
        class="c2"
      >
        Gympass Black for employees with Buddha Spa + Reebook included
      </h3>
      <strong
        class="c3"
      >
        U$ 99.90
      </strong>
      <span
        class="c4"
      >
        /month
      </span>
    </div>
    <footer
      class="c5"
    >
      <button
        class="c6"
      >
        should have buttons
      </button>
    </footer>
  </section>
</div>
`;

exports[`<PlanCard /> Snapshots should match snapshot with secondary PlanCard 1`] = `
.c6 {
  box-sizing: border-box;
  outline: none;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  cursor: pointer;
  width: auto;
  height: 48px;
  padding-left: 24px;
  padding-right: 24px;
  background-color: #F46152;
  border: 2px solid #F46152;
  border-radius: 9999px;
  color: #FFFFFF;
  font-size: 14px;
  font-weight: 700;
}

.c6:not([disabled]):hover,
.c6:not([disabled]):focus {
  box-shadow: 0 4px 8px rgba(244,97,82,0.45);
}

.c6:active {
  background-color: #F78D82;
  border-color: #F78D82;
  color: #FFFFFF;
}

.c6:disabled {
  background-color: #DBDBE8;
  border-color: #DBDBE8;
  color: #9898A6;
  cursor: not-allowed;
}

.c0 {
  padding: 16px  16px  16px  16px;
  border-radius: 8px;
  background-color: #FFFFFF;
  box-shadow: 0 2px 6px rgba(152,152,166,0.25);
}

.c1 {
  max-width: 280px;
  width: 100%;
  color: #FFFFFF;
  background-color: #652CDB;
}

.c2 {
  display: -webkit-box;
  height: 40px;
  margin: 8px 0 12px;
  font-size: 14px;
  font-weight: 400;
  color: inherit;
  -webkit-line-clamp: 2;
  overflow: hidden;
  text-overflow: ellipsis;
  text-overflow: -o-ellipsis-lastline;
}

.c3 {
  display: block;
  margin-top: 12px;
  font-size: 20px;
  font-weight: 700;
}

.c4 {
  display: block;
  margin-bottom: 8px;
  font-size: 14px;
  font-weight: 400;
}

.c5 {
  display: grid;
  grid-gap: 8px;
  padding-top: 8px;
}

<div>
  <section
    class="c0 c1"
  >
    <header
      class=""
    >
      Gympass PlanCard
    </header>
    <div
      class=""
    >
      <h3
        class="c2"
      >
        Gympass Black for employees with Buddha Spa + Reebook included
      </h3>
      <strong
        class="c3"
      >
        U$ 99.90
      </strong>
      <span
        class="c4"
      >
        /month
      </span>
    </div>
    <footer
      class="c5"
    >
      <button
        class="c6"
      >
        should have buttons
      </button>
    </footer>
  </section>
</div>
`;

exports[`<PlanCard /> Snapshots should match snapshot with tertiary PlanCard 1`] = `
.c6 {
  box-sizing: border-box;
  outline: none;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  cursor: pointer;
  width: auto;
  height: 48px;
  padding-left: 24px;
  padding-right: 24px;
  background-color: #F46152;
  border: 2px solid #F46152;
  border-radius: 9999px;
  color: #FFFFFF;
  font-size: 14px;
  font-weight: 700;
}

.c6:not([disabled]):hover,
.c6:not([disabled]):focus {
  box-shadow: 0 4px 8px rgba(244,97,82,0.45);
}

.c6:active {
  background-color: #F78D82;
  border-color: #F78D82;
  color: #FFFFFF;
}

.c6:disabled {
  background-color: #DBDBE8;
  border-color: #DBDBE8;
  color: #9898A6;
  cursor: not-allowed;
}

.c0 {
  padding: 16px  16px  16px  16px;
  border-radius: 8px;
  background-color: #FFFFFF;
  box-shadow: 0 2px 6px rgba(152,152,166,0.25);
}

.c1 {
  max-width: 280px;
  width: 100%;
  color: #FFFFFF;
  background-color: #2B9486;
}

.c2 {
  display: -webkit-box;
  height: 40px;
  margin: 8px 0 12px;
  font-size: 14px;
  font-weight: 400;
  color: inherit;
  -webkit-line-clamp: 2;
  overflow: hidden;
  text-overflow: ellipsis;
  text-overflow: -o-ellipsis-lastline;
}

.c3 {
  display: block;
  margin-top: 12px;
  font-size: 20px;
  font-weight: 700;
}

.c4 {
  display: block;
  margin-bottom: 8px;
  font-size: 14px;
  font-weight: 400;
}

.c5 {
  display: grid;
  grid-gap: 8px;
  padding-top: 8px;
}

<div>
  <section
    class="c0 c1"
  >
    <header
      class=""
    >
      Gympass PlanCard
    </header>
    <div
      class=""
    >
      <h3
        class="c2"
      >
        Gympass Black for employees with Buddha Spa + Reebook included
      </h3>
      <strong
        class="c3"
      >
        U$ 99.90
      </strong>
      <span
        class="c4"
      >
        /month
      </span>
    </div>
    <footer
      class="c5"
    >
      <button
        class="c6"
      >
        should have buttons
      </button>
    </footer>
  </section>
</div>
`;
